lec11-modelling.Rmd

```diff
@@ -0,0 +1,527 @@
+maybe we know that $y$ has a lot of error in the measurement or is difficult to
+measure, so we want to derive a formula in order to predict the variable based
+on more accurately measured variables. In R we can run linear regression either
+using `lm` or using `glm`. We'll use `glm` since it can be expanded to other
+types of analyses. First, let's load up a dataset.
+
+```{r, message=FALSE}
+portal <- read_csv("portal_data.csv")
+```
+
+Now, let's fit a model to the data. Let's say we want to see the role that sex
+has on weight in various species in the `portal` dataset. (`gaussian` is stating 
+that `weight` is a continuous variable and that you assume the error terms have
+a gaussian, or normal, distribution.)
+
+```{r}
+fit1 <- glm(weight ~ sex, data = portal, family = gaussian)
+summary(fit1)
+```
+
+There's a lot of results and information contained within `fit1`, which
+`summary()` extracts and presents in a fairly nice format. However, a lot of
+this information we aren't really interested in. So let's extract only what we
+want, using the `broom` package. (get it, broom to clean up?)
+
+```{r}
+tidy(fit1)
+```
+
+Much nicer! Ok, so what does this actually mean? Let's go back to the equation
+and convert the above into the formula:
+
+$$ weight = Intercept + (sexM \times b) $$
+
+Let's substitute in the numbers. The intercept, is well, the intercept in the
+data above. The $b$ is the estimate for `sex`.
+
+$$ weight = 42.17 + (sexM \times 0.825) $$
+
+We can use this equation to estimate the value of weight for a specific sex. But
+first, let's make sure this is working correctly. Let's find out the mean weight
+by sex.
+
+```{r}
+portal %>% 
+    filter(!is.na(sex)) %>% 
+    group_by(sex) %>% 
+    summarize(mean_wt = mean(weight, na.rm = TRUE)) 
+```
+
+Ok, let's compare to the equation. The variable `sexM` is 1 for male and 0 for
+female. So if we want to calculate the weight of female, we have to set `sexM`
+as 0 in the equation. Let's do it:
+
+```{r}
+42.17 + (0 * 0.825)
+```
+
+Same as above! And for males (equal to 1):
+
+```{r}
+42.17 + (1 * 0.825)
+```
+
+Good, it matches what we calculated. Why is this useful? Well, we can add more
+terms to the GLM equation. This allows us to determine how a variable influences
+the $y$ when other variables are held constant and is known as *multiple linear
+regression*. Since weight is probably also dependent on `hindfoot_length`, let's
+add that to the model and put the values into the equation.
+
+```{r}
+fit2 <- glm(weight ~ sex + hindfoot_length, data = portal)
+tidy(fit2)
+```
+
+$$ weight = -32.09 + (sexM \times -1.72) + (hindfoot\_length \times 2.56)$$
+
+Hmm, that's interesting... is a *negative* intercept... How can weight be
+negative? Well, that's because the intercept is calculated by setting *both* sex
+and hindfoot length to zero! But is it physically even possible that a hindfoot be
+non-existant?? No. But math doesn't care what is physically possible or not.
```

Comment:
non-exist**e**nt

Reply:
lol `math doesn't care what is physically possible or not`

and that's how we got black holes. Friggin math.